src/halmos/__main__.py

```diff
@@ -415,6 +421,52 @@ def get_state_id(ex: Exec) -> bytes:
     return snapshot_state(ex, include_path=True).unwrap()
 
 
+def run_target_function(
+    args: HalmosConfig,
+    ex: Exec,
+    addr: Address,
+    abi: dict,
+    fun_info: FunctionInfo,
+    tx_origin: Address,
+    msg_sender: Address,
+    msg_value: Word,
+    msg_sender_cond: BoolRef | None = None,
+) -> Iterator[Exec]:
+    try:
+        # initialize symbolic execution environment
+        sevm = SEVM(args, fun_info)
+        solver = mk_solver(args)
+        path = Path(solver)
+        path.extend_path(ex.path)
+
+        # prepare calldata and dynamic parameters
+        calldata, dyn_params = mk_calldata(
+            abi, fun_info, args, new_symbol_id=ex.new_symbol_id
+        )
+        path.process_dyn_params(dyn_params)
+
+        # add (optional) constraints on msg_sender
+        if msg_sender_cond:
```

Comment:
doesn't this give an error if `msg_sender_cond` is an actual BoolRef? does it mean this is never called?

```suggestion
        if msg_sender_cond is not None:
```

Reply:
indeed adding a call to `targetSender(x)` in one of the invariant tests results in this error:

```
ERROR    run_target_contract 2863267842 inc(uint256): Z3Exception: Symbolic expressions cannot be cast to concrete Boolean values.                                                                                                                                                                                      
```

Reply:
oh good catch!

Reply:
https://github.com/a16z/halmos/pull/487/commits/1cd0c3a27260782d19c38ff84c320222484c8142